### PR TITLE
Adds pineapples to botany, and Hawaiian pizza

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis, Cosmic, Fwoosh, and epochayur
add: Added pineapples to botany, and a recipe for Hawaiian pizza.
/:cl:

By request of @CosmicScientist. Sprites provided by them, @FantasticFwoosh, and @epochayur.

Pineapples by themselves don't have any unique traits, but they can be sliced into pieces, and those slices can be used to make Hawaiian pizza (you also need two cutlets, for ham.)

Also added a Hawaiian pizza box to the pizza crate.

### CODE REVIEW

**Comment by praisenarsie** [comment]
holy shit those pineapple slices are giant

**Comment by Xhuis** [comment]
Giantly delicious.

**Comment by Cyberboss** [comment]
Eating one pizza and loving it and then eating another and hating it seems weird

**Comment by DaedalusGame** [comment]
How could you betray me like this

**Comment by arsserpentarium** [comment]
Maybe add some carbon in genes?

**Comment by vuonojenmustaturska** [comment]
Why would you have pizza with just pineapple on it, though? Pineapple on pizza only really shines when combined with something like really salty ham, or pepperoni.

**Comment by DaedalusGame** [comment]
ok i think we need to clean up maintainers

**Comment by Anonmare** [comment]
Wouldn't it make more sense to have players spawn with a 50% chance of loving/hating Pineapple pizza?

**Comment by Xhuis** [comment]
@Cyberboss The pineapple giveth and the pineapple taketh.
@Anonmare I'm not sure if that's possible without weird static vars etc., but I can take a look. Someone suggested I add pizza preferences to the character setup, and I might just do that. 🤔

**Comment by DaedalusGame** [comment]
Base it on species, catgirls can settle for pineapple pizza for all i care

**Comment by epochayur** [comment]
petition to repoban xhuis now

**Comment by Durkel** [comment]
Give them a mutation that will turn them into pineapple grenades so I can watch people complain on the forums about botany again.

**Comment by DaedalusGame** [comment]
yeah make them produce n2o tbh

**Comment by CosmicScientist** [comment]
YES. THE PINEAPPLE COALITION IS UNSTOPPABLE!

**Comment by Xhuis** [comment]
@Cyberboss Would putting `DEVIANT` into the species' `liked_food` or `disliked_food` at random on `/species/New()` be shitcodey, or good?

**Comment by Nabski89** [comment]
Jokepost: Make them able to mutate into pineapple grenade plants.

**Comment by uraniummeltdown** [comment]
Sprites are kinda bad
Look at https://github.com/ParadiseSS13/Paradise/pull/8014 for inspiration

File: code/modules/food_and_drinks/food/snacks_pizza.dm [hunk 1]
```diff
@@ -139,6 +139,31 @@
...
+	icon_state = "pineapplepizza"
+	slice_path = /obj/item/reagent_containers/food/snacks/pizzaslice/pineapple
+	bonus_reagents = list("nutriment" = 6, "vitamin" = 6)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pineapples" = 2)
+	foodtype = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/reagent_containers/food/snacks/pizza/pineapple/Initialize()
+	. = ..()
+	if(prob(50))
+		foodtype |= GROSS //either you love pineapple pizza, or you hate pineapple pizza.
```
Comment (ChangelingRain) [inline code/modules/food_and_drinks/food/snacks_pizza.dm]: this isn't how flavour works, just remove this

if you hate pineapple pizza fucking roleplay it out

**Review by ChangelingRain — CHANGES_REQUESTED** [review]

**Comment by arsserpentarium** [comment]
Pineapple grenade should have gaseous decomposition

**Comment by epochayur** [comment]
TIL pineapples don't grow on trees, that's pretty fucked up

**Comment by Xhuis** [comment]
I didn't make them, don't ask me!

File: code/modules/food_and_drinks/food/snacks_pizza.dm (hunk 1 above)
Comment (Xhuis) [reply code/modules/food_and_drinks/food/snacks_pizza.dm]: is a single joke about the controversy of pineapple pizza really terrible to implement in code

**Comment by epochayur** [comment]
if i might put a horse in this race; 
image

tried to make it not fuckhuge and a little more fitting with the looks of the rest of the food items
https://mega.nz/#!ggMz2IjR

**Comment by Xhuis** [comment]
Apparently I need a decryption key to access that file...?

**Comment by epochayur** [comment]
sorry, I'm retarded, try the edited link

**Comment by Xhuis** [comment]
implemented, thank you!

**Comment by Iatots** [comment]
the meme is pineapple **on** pizza, like hawaiian style, not "pineapple pizza"

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_pizza.dm [hunk 2]
```diff
@@ -77,3 +77,14 @@
...
 	subcategory = CAT_PIZZA
+
+/datum/crafting_recipe/food/pineapplepizza
+	name = "Pineapple pizza"
+	reqs = list(
+		/obj/item/reagent_containers/food/snacks/pizzabread = 1,
+		/obj/item/reagent_containers/food/snacks/pineappleslice = 3,
+		/obj/item/reagent_containers/food/snacks/cheesewedge = 1,
+		/obj/item/reagent_containers/food/snacks/grown/tomato = 1
+	)
```
Comment (DaxDupont) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_pizza.dm]: where the fuck is the cutlets for the ham substitute

**Comment by optimumtact** [comment]
just add a pineapple lovers trait that randomises on roundstart and gives you a new foodgroup for pineapple pizzas

**Comment by Kraseo** [comment]
hawaiian style is canadian

**Comment by Bombpants** [comment]
This PR is amazing

**Comment by Bawhoppen** [comment]
Master hour sprites

**Comment by Bawhoppen** [comment]
Actually the sprite is acceptable I retract my statement

**Comment by Xhuis** [comment]
@optimumtact that's doable, but how would I assign it? just whenever a human is given a mind?

**Comment by optimumtact** [comment]
do it in the species gain

**Comment by Bawhoppen** [comment]
Add it as a pref ofc

**Comment by optimumtact** [comment]
you will die baw

**Comment by Bawhoppen** [comment]
We need a minor prefs system so oranges won't be mad about this stuff

**Comment by optimumtact** [comment]
We do actually need something along those lines

**Comment by Supermichael777** [comment]
Test out that deviant trait so we can add yeast extract

**Comment by ChangelingRain** [comment]
please do not randomize liking pineapple on pizza with code holy shit
let people decide if their character likes pineapple on pizza on their own

File: code/modules/food_and_drinks/food/snacks_pizza.dm (hunk 1 above)
Comment (ChangelingRain) [reply code/modules/food_and_drinks/food/snacks_pizza.dm]: yes, it is terrible, because people don't get a choice about a character quirk.

**Comment by Anonmare** [comment]
if you have the Syndicate faction, you should always like Pineapple Pizza. Everyone knows the -bad- good guys like it.

**Comment by Xhuis** [comment]
fiiiine, since you people hate fun, I'll remove the randomized liking of it. I'm kind of curious about coding a minor preferences system now, but I'm not sure how/where/what would be put into it.

**Comment by optimumtact** [comment]
it'll end up in game prefs like everything else

File: code/modules/food_and_drinks/food/snacks_pizza.dm [hunk 3]
```diff
@@ -139,6 +139,22 @@
...
 	foodtype = GRAIN | VEGETABLES | DAIRY
 
+/obj/item/reagent_containers/food/snacks/pizza/pineapple
+	name = "\improper Hawaiian pizza"
+	desc = "The pizza equivalent of Einstein's riddle."
+	icon_state = "pineapplepizza"
+	slice_path = /obj/item/reagent_containers/food/snacks/pizzaslice/pineapple
+	bonus_reagents = list("nutriment" = 6, "vitamin" = 6)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pineapples" = 2, "ham" = 2)
+	foodtype = GRAIN | VEGETABLES | DAIRY
```
Comment (ChangelingRain) [inline code/modules/food_and_drinks/food/snacks_pizza.dm]: no fruit?

Comment (Xhuis) [reply code/modules/food_and_drinks/food/snacks_pizza.dm]: Whoop, oversight

**Review by ChangelingRain — APPROVED** [review]

File: code/modules/food_and_drinks/food/snacks_pizza.dm [hunk 4]
```diff
@@ -139,6 +139,23 @@
 	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
 	foodtype = GRAIN | VEGETABLES | DAIRY
 
+/obj/item/reagent_containers/food/snacks/pizza/pineapple
+	name = "\improper Hawaiian pizza"
```
Comment (Iamgoofball) [inline code/modules/food_and_drinks/food/snacks_pizza.dm]: *space hawaiian

**Pull request merged**